docs/guides/typescript.md

```diff
@@ -0,0 +1,225 @@
+- Projects transitioning to Platformatic without rewriting build processes
+
+## Solution Overview
+
+This guide walks you through migrating a standalone TypeScript application to Watt while keeping ts-node. We'll cover:
+
+1. Understanding your current standalone setup
+2. Configuring Watt to use ts-node with `execArgv`
+3. Adjusting your development and production workflows
+4. Handling both ESM and CommonJS module systems
+
+## Your Current Standalone Setup
+
+Before migrating, let's look at a typical standalone TypeScript application structure:
+
+**Directory structure:**
+
+```
+my-app/
+├── src/
+│   └── index.ts
+├── package.json
+├── tsconfig.json
+└── node_modules/
+```
+
+**package.json:**
+
+```json
+{
+  "name": "my-app",
+  "main": "dist/index.js",
+  "type": "module",
+  "scripts": {
+    "dev": "ts-node src/index.ts",
+    "build": "tsc",
+    "start": "node dist/index.js"
+  },
+  "dependencies": {
+    "fastify": "^5.0.0"
+  },
+  "devDependencies": {
+    "@types/node": "^22.0.0",
+    "ts-node": "^10.9.2",
+    "typescript": "^5.9.3"
+  }
+}
+```
+
+**tsconfig.json:**
+
+```json
+{
+  "compilerOptions": {
+    "target": "ES2022",
+    "module": "ES2022",
+    "moduleResolution": "node",
+    "outDir": "dist",
+    "strict": true
+  }
+}
+```
+
+**src/index.ts:**
+
+```typescript
+import { createServer } from 'node:http'
+
+const server = createServer((req, res) => {
+  res.writeHead(200, { 'Content-Type': 'application/json' })
+  res.end(JSON.stringify({ message: 'Hello World' }))
+})
+
+server.listen(3000, () => {
+  console.log('Server listening on port 3000')
+})
+```
+
+You run `npm run dev` to start development with ts-node, and everything works. Now let's migrate this to Watt.
+
+## Migration Steps
+
+This migration uses the standalone Node.js capability approach for a simpler single-application setup.
+
+### Step 1: Install Dependencies
+
+```bash
+npm install wattpm @platformatic/globals @platformatic/node
+npm install -D @platformatic/tsconfig
+```
+
+### Step 2: Create Watt Configuration
+
+**Create `watt.json` in your project root (for ESM applications):**
+
+```json
+{
+  "$schema": "https://schemas.platformatic.dev/@platformatic/node/3.25.0.json",
+  "runtime": {
+    "application": {
+      "execArgv": ["--no-warnings", "--loader", "ts-node/esm"],
+      "commands": {
+        "build": "tsc"
+      }
+    }
+  },
+  "node": {
+    "skipBuildInDevelopment": true
+  }
+}
+```
+
+Key configuration details:
+
+- Uses `@platformatic/node` schema
+- `execArgv` passes ts-node ESM loader to Node.js process
+- `skipBuildInDevelopment: true` ensures TypeScript runs directly in dev mode
+
+**Note:** For CommonJS applications, use different `execArgv`. See the [Handling CommonJS Applications](#handling-commonjs-applications) section below.
+
+### Step 3: Update Package.json
+
+Keep your application structure mostly the same:
+
+```json
+{
+  "name": "my-app",
+  "main": "src/index.ts",
+  "type": "module",
```

Comment:
Assume the reader does not know anything about esm, cjs or anything in between. Assume that the original application is faux esm, not esm.